addons/base_vat/models/res_partner.py

```diff
@@ -222,7 +222,16 @@ def _build_vat_error_message(self, country_code, wrong_vat, record_label):
         else:
             company = self.env.company
 
-        vat_label = company.country_id.vat_label if company.country_id else "Tax ID"
+        vat_label = _("VAT")
+        if country_code:
+            partner_country = self.env['res.country'].search([('code', '=', country_code.upper())], limit=1)
+            # If the partner and company both are in the same country, vat_label only
+            if partner_country == company.country_id and partner_country.vat_label:
+                vat_label = partner_country.vat_label
+            # If the partner and company both are in different country, VAT with vat_label like "VAT (vat_label)"
+            elif partner_country.vat_label and partner_country.vat_label != vat_label:
+                vat_label += "(" + partner_country.vat_label + ")"
```

Comment:
```suggestion
        if not country_code or country_code == company.country_id.code: 
           vat_label = company.country_id.vat_label or vat_label
       elif country_code and country_code != company.country.code and company_id.country.vat_label not in [False, "", "vat"]: #Not sure yet
            vat_label  += " (" + (company.country_id.vat_label or vat_label) + ")"   
```
@chklop Right now, the name for the VAT is based on the current company.  I don't want to make it depend on the country of the partner.  Even if this is only for an error message, if I am in Switzerland and I send something to Ecuador they should not be talking about RUC in Switzerland e.g.

Reply:
in case of the country code and company country code is different then we must need to browse or read the vat label from `res.country`

Reply:
@jva-odoo It seems to me that the vat_label on the country of the partner should have any role at all.

Reply:
the role is there when company's country and partner's country is different

example the company country is Argentina
====================================

If Partner Country and Company Country are different then (here partner country is India)

- The VAT(GSTIN) number xxx for Partner yyy does not seem valid

If both are the same then

- The CUIT number xxx for Partner yyy does not seem valid


another example is the company country is India
======================================

If Partner Country and Company Country are different then (here partner country is Argentina)

- The VAT(CUIT) number xxx for Partner yyy does not seem valid

If both are the same then

- The GSTIN number xxx for Partner yyy does not seem valid